homeassistant/components/config/device_registry.py

```diff
@@ -72,22 +59,29 @@ def websocket_list_devices(hass, connection, msg):
         run_immediately=True,
     )
 
-    websocket_api.async_register_command(
-        hass, WS_TYPE_LIST, websocket_list_devices, SCHEMA_WS_LIST
-    )
-    websocket_api.async_register_command(
-        hass, WS_TYPE_UPDATE, websocket_update_device, SCHEMA_WS_UPDATE
-    )
+    websocket_api.async_register_command(hass, websocket_list_devices)
+    websocket_api.async_register_command(hass, websocket_update_device)
     websocket_api.async_register_command(
         hass, websocket_remove_config_entry_from_device
     )
     return True
 
 
 @require_admin
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "config/device_registry/update",
+        vol.Optional("area_id"): vol.Any(str, None),
+        vol.Required("device_id"): str,
+        # We only allow setting disabled_by user via API.
+        # No Enum support like this in voluptuous, use .value
+        vol.Optional("disabled_by"): vol.Any(DeviceEntryDisabler.USER.value, None),
+        vol.Optional("name_by_user"): vol.Any(str, None),
+    }
+)
 @callback
 def websocket_update_device(hass, connection, msg):
```

Comment:
Could you add type hints here at the same time?
```suggestion
def websocket_update_device(
    hass: HomeAssistant,
    connection: websocket_api.ActiveConnection,
    msg: dict[str, Any],
) -> None:
```